### ISSUE TITLE
T9 Support

### ISSUE THREAD

**Issue by maverick74** [SimpleMobileTools/Simple-Keyboard#20]
Would be great to have support for the T9 Keyboard layout and text prediction.

Could you please consider adding it?

**Comment by maverick74** [comment]
@tibbi this is surely a HUGE task...

is there any hope for this any time "soon"?

**Comment by tibbi** [comment]
definitely not anytime soon, maybe never actually

**Comment by TwiggyWan** [comment]
> definitely not anytime soon, maybe never actually

hello, is it not doable with current app, or 'just' too much work?

AFAIK there is only 'smart keyboard pro' which provides this feature.. not opensource, privacy concerns.

**Comment by TwiggyWan** [comment]
and yet burner phones from the early 2000s managed it offline and those did not have huge storage space

**Comment by maverick74** [comment]
> AFAIK there is only 'smart keyboard pro' which provides this feature.. not opensource, privacy concerns.

TouchPal also supports this and is free (but has many privacy concerns as well)

**Comment by maverick74** [comment]
> I dont think this can be done in an offline app, I would have to add dictionaries of all supported languages that would drastically increase the app size

Maybe having dictionaries as add-ons the user could install. AnySoftKeyboard (and others) work this way.
the user installs the languages he wishes!

**Comment by tibbi** [comment]
> > I dont think this can be done in an offline app, I would have to add dictionaries of all supported languages that would drastically increase the app size
> 
> Maybe having dictionaries as add-ons the user could install. AnySoftKeyboard (and others) work this way. the user installs the languages he wishes!

that sounds way too complicated to me, I dont want to go that way

**Comment by maverick74** [comment]
> @tibbi I understand.
> 
> @maverick74 I guess we can use touchpal and use netguard/tracker control to disable the internet then.

Yeah, that's what I do!!! But I would still prefer an open-source keyboard!

**Comment by TwiggyWan** [comment]
This would be a nice side project..

That's why I asked @tibbi if that can somehow be plugged into what's existing in simple keyboard, or if it's so complicated that it'd need a full rewrite..

**Comment by GiorgosXou** [comment]
@TwiggyWan  @maverick74  [That's how I use it too](https://github.com/openboard-team/openboard/issues/379#issuecomment-1527223453) but the sad reality is that touchpal has stopped upgrading, meaning no more emojis ☹️

**Comment by maverick74** [comment]
> @TwiggyWan @maverick74 [That's how I use it too](https://github.com/openboard-team/openboard/issues/379#issuecomment-1527223453) but the sad reality is that touchpal has stopped updating, meaning no more emojis too ☹️

I `m very happy with Traditional T9 ( https://github.com/sspanak/tt9 )

**Comment by a-semmar** [comment]
> > @TwiggyWan @maverick74 [That's how I use it too](https://github.com/openboard-team/openboard/issues/379#issuecomment-1527223453) but the sad reality is that touchpal has stopped updating, meaning no more emojis too ☹️
> 
> I `m very happy with Traditional T9 ( https://github.com/sspanak/tt9 )

The description of TT9 izzyondroid page  says 'TT9 is not usable on touchscreen-only devices.' . So how does it work for you?

**Comment by maverick74** [comment]
> > > @TwiggyWan @maverick74 [That's how I use it too](https://github.com/openboard-team/openboard/issues/379#issuecomment-1527223453) but the sad reality is that touchpal has stopped updating, meaning no more emojis too ☹️
> > 
> > 
> > I `m very happy with Traditional T9 ( https://github.com/sspanak/tt9 )
> 
> The description of TT9 izzyondroid page says 'TT9 is not usable on touchscreen-only devices.' . So how does it work for you?

Yes, but on the site I sent also mentions
" A hardware keypad or a keyboard. For touchscreen-only devices, an on-screen keypad can be enabled in the Settings. "

its a beta feature, however!

**Comment by eddysnjack** [comment]
Until few months ago I was still using touchpal (restrict the internet on it and fingers crossed) now I had to switch to a new phone and it is using android 13 which touchpal not working with it. Idk what to do. I'm just trying to understand how t9 works, how can I implement it etc. And seems like it may took years to reach the experience touchpal provide. I just can't bare qwerty, it's seriously unbareable. If anyone here willing to create touchpal t9 clone I'd be happy to help. I really want that t9 keyboard and of course safety.

**Comment by maverick74** [comment]
> Until few months ago I was still using touchpal

as I mentioned above, I used touchpal also and I am now perfectly happy with tt9 ( https://github.com/sspanak/tt9 )! 


There are one or two features I miss (such as delete a whole word at once) but I can live without it for now

**Comment by eddysnjack** [comment]
> > Until few months ago I was still using touchpal
> 
> as I mentioned above, I used touchpal also and I am now perfectly happy with tt9 ( https://github.com/sspanak/tt9 )! 
> 
> 
> There are one or two features I miss (such as delete a whole word at once) but I can live without it for now

Yeah I tried tt9 but it didn't make the cut tbh. I appreciate to effort and open sourceness tho, as always.

**Comment by GiorgosXou** [comment]
@eddysnjack [SimpleMobileTools got sold to the Israeli publisher ZipoApps](https://github.com/SimpleMobileTools/General-Discussion/issues/241). So, you might be interested raising this issue at *(the fork)* [FossifyOrg/Keyboard](https://github.com/FossifyOrg/Keyboard) *(because I don't think SimpleMobileTools will remain open source anymore from now on)*

**Comment by maverick74** [comment]
> Yeah I tried tt9 but it didn't make the cut tbh.

Out of pure curiosity: what were the blockers for you?
